File: frontend/lib/data/models/festival.dart
import 'package:json_annotation/json_annotation.dart';

part 'festival.g.dart';

@JsonSerializable()
class Festival {
	final int id;
	final String name;
	final String? nameLocal;
	final String? nameEnglish;
	final int type;
	final int? regionId;
	final int? ethnicityId;
	final String? startDate;
	final String? endDate;
	final int duration;
	final int isLunar;
	final int isHoliday;
	final int? holidayDays;
	final String? description;
	final String? history;
	final String? customs;
	final String? food;
	final String? clothing;
	final String? activities;
	final String? imageUrl;
	final String? videoUrl;
	final int status;

	Festival({
		required this.id,
		required this.name,
		this.nameLocal,
		this.nameEnglish,
		required this.type,
		this.regionId,
		this.ethnicityId,
		this.startDate,
		this.endDate,
		this.duration = 1,
		this.isLunar = 0,
		this.isHoliday = 0,
		this.holidayDays,
		this.description,
		this.history,
		this.customs,
		this.food,
		this.clothing,
		this.activities,
		this.imageUrl,
		this.videoUrl,
		this.status = 1,
	});

	factory Festival.fromJson(Map<String, dynamic> json) => _$FestivalFromJson(json);
	Map<String, dynamic> toJson() => _$FestivalToJson(this);

	String get typeText {
		switch (type) {
			case 1: return '法定假日';
			case 2: return '传统节日';
			case 3: return '民族节日';
			case 4: return '节气';
			case 5: return '纪念日';
			default: return '其他';
		}
	}
}

File: backend/src/main/java/com/regional/calendar/entity/Festival.java
package com.regional.calendar.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("festival")
public class Festival {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String nameLocal;

    private String nameEnglish;

    private Integer type;

    private String subType;

    private Long regionId;

    private String regionName;

    private Long ethnicityId;

    private String ethnicityName;

    private String startDate;

    private String endDate;

    private Integer duration;

    private Integer isLunar;

    private Integer lunarMonth;

    private Integer lunarDay;

    private Integer isHoliday;

    private Integer holidayDays;

    private Integer isSolarTerm;

    private BigDecimal solarTermAngle;

    private String description;

    private String origin;

    private String history;

    private String customs;

    private String taboos;

    private String food;

    private String foodImages;

    private String clothing;

    private String clothingImages;

    private String activities;

    private String symbols;

    private String poems;

    private String songs;

    private String imageUrl;

    private String iconUrl;

    private String bannerUrl;

    private String videoUrl;

    private String gallery;

    private String color;

    private String tags;

    private Long viewCount;

    private Long favoriteCount;

    private Long shareCount;

    private Integer sortOrder;

    private Integer isHot;

    private Integer isRecommended;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
